File: java/L200/LeetCode0231.java
package L200;

public class LeetCode0231 {
    /**
     * 如果是 2 的幂，则二进制只有一个 1，其余为 0；
     */
    public boolean isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        }
        return (n & (n - 1)) == 0;
    }
}

File: cpp/L200/LeetCode0231.cc
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n <= 0) {
            return false;
        } 
        int ans{0};
        for (int i = 0;i < 31;++i) {
            if (((n >> i) & 1) == 1) {
                ++ans;
            }
        }
        return ans == 1;
    }
};